services/QuillLMS/app/models/topic.rb

```diff
@@ -64,6 +68,9 @@ def level_three?
   end
 
   private def valid_parent_structure?
-    level_two? ? level_three_parent? : no_parent?
+    return level_two_parent? if level_one?
+    return level_three_parent? if level_two?
+
+    no_parent?
```

Comment:
Is it the case that `no_parent?` `Topic`s are only valid if they're `level_three?`?  Should we include that as part of this validation?